nimpy.nim

```diff
@@ -957,6 +1016,142 @@ proc pyObjToNimSeq[T](o: PPyObject, v: var seq[T]) =
         pyObjToNim(pyLib.PyList_GetItem(o, i), v[i])
         # PyList_GetItem # No DECREF. Returns borrowed ref.
 
+
+proc pyObjToNimTab[T; U](o: PPyObject, tab: var Table[T, U]) =
+    ## call this either:
+    ## - if you want to check whether T and U are valid types for
+    ##   the python dict (i.e. to check whether all python types
+    ##   are convertible to T and U)
+    ## - you know the python dict conforms to T and U and you wish
+    ##   to get a correct Nim table from that
+    tab = initTable[T, U]()
+    let
+        sz = int(pyLib.PyDict_Size(o))
+        ks = pyLib.PyDict_Keys(o)
+        vs = pyLib.PyDict_Values(o)
+    for i in 0 ..< sz:
+        var
+            k: T
+            v: U
+        pyObjToNim(pyLib.PyList_GetItem(ks, i), k)
+        pyObjToNim(pyLib.PyList_GetItem(vs, i), v)
+        # PyList_GetItem # No DECREF. Returns borrowed ref.
+        tab[k] = v
+    decref ks
+    decref vs
+
+proc `%`(o: PPyObject): JsonNode
+proc `%`(o: PyObject): JsonNode
+
+template getVal(pyVal: PPyObject, actions: untyped) {.dirty.} =
```

Comment:
I suppose you dont need dirty here.

Reply:
Good point, I don't if I just inject `x`.